### ISSUE TITLE
Webpage has no Data after update

### ISSUE THREAD

**Issue by JobbeDeluxe** [pi-hole/pi-hole#1832]
**In raising this issue, I confirm the following:** `{please fill the checkboxes, e.g: [X]}`

- [X] I have read and understood the [contributors guide](https://github.com/pi-hole/pi-hole/blob/master/CONTRIBUTING.md).
- [X] The issue I am reporting can be *replicated*.
- [] The issue I am reporting isn't a duplicate (see [FAQs](https://github.com/pi-hole/pi-hole/wiki/FAQs), [closed issues](https://github.com/pi-hole/pi-hole/issues?utf8=%E2%9C%93&q=is%3Aissue%20is%3Aclosed%20), and [open issues](https://github.com/pi-hole/pi-hole/issues)).

**How familiar are you with the the source code relevant to this issue?:**

`{5}`

---
**Expected behaviour:**

`{See all data on the webpage}`

**Actual behaviour:**

`{i can only see the "Recent Queries" all other status are blank or in endles loading.}`

**Steps to reproduce:**

`{Start the webpage on /html/admin/index.php }`

**Debug token provided by [uploading `pihole -d` log](https://discourse.pi-hole.net/t/the-pihole-command-with-examples/738#debug):**

`{hb0q4gd76s}`

**Troubleshooting undertaken, and/or other relevant information:**

`{I have installed an update than it appears. i have tryed "pihole -r" i have flushed the log with "pihole -flush" i have uninstalled pihole and reinstalled it. Nothing changed.}`
![auswahl_001](https://user-images.githubusercontent.com/20384365/33804003-9610fc1c-dd9c-11e7-9b6d-dec861959d26.png)

**Comment by DL6ER** [comment]
This is something that keeps coming up after each release for a number of users and is almost always related to caching in your local browser. These issues fix themselves magically after some time. To fix it now, try `Ctrl + F5` or `Crtl + Shift + R` (or whatever the Mac equivalent ist).

**Comment by echaritos** [comment]
Can confirm this is client based. Ctrl + F5 fixed my issue

**Comment by JobbeDeluxe** [comment]
That will not work either. I tried from many browsers and devices. 
![screenshot_2017-12-10-12-07-17](https://user-images.githubusercontent.com/20384365/33804323-bdc15d00-dda2-11e7-8f44-2593590228d1.png)

**Comment by DL6ER** [comment]
It is very strange that it loads partially for you (as all data is requested from the same source). Can you confirm that you have the most recent version of `FTL` ?

```
pihole-FTL version
```
Maybe you can try a restart of `FTL` in case the binary got updated but the service wasn't restarted
```
sudo service pihole-FTL restart
```

**Comment by JobbeDeluxe** [comment]
:~ $ pihole-FTL version
vDev-437af07

I have restarted the ftl service and the server as well. No change.
Bevor the update it was shown correctly.

**Comment by echaritos** [comment]
Can you confirm that you have the same problem when viewing the webpage on a PC/Mac? The screenshot seems to be from a mobile device

**Comment by JobbeDeluxe** [comment]
The first is from a laptop.

**Comment by JobbeDeluxe** [comment]
But it seems to be a client problem. I just installed opera to be sure....
And there it will be displayed correctly 
![screenshot_2017-12-10-12-32-20](https://user-images.githubusercontent.com/20384365/33804480-456ace64-dda6-11e7-9f28-9ebefc2289c8.png)

**Comment by DL6ER** [comment]
Hmm, could you try the `Request Desktop Site` option on your Android device?

Which browser are you using on your Desktop? We test Vivaldi, Firefox, Chrome on Linux (partially on Win) as well as Safari on Mac during development, all in their most recent releases (regular releases, not development releases) and haven't seen such an issue.

**Comment by JobbeDeluxe** [comment]
I tried Chrome on Linux and android.
The same problem on Android with desktop view. 
![screenshot_2017-12-10-12-39-43](https://user-images.githubusercontent.com/20384365/33804538-5bcf8f04-dda7-11e7-992c-35848a44d1c1.png)

**Comment by DL6ER** [comment]
Okay, can you please open the developers tools on Chrome/Linux (press `F12`) and look for errors in the `Console`?

**Comment by JobbeDeluxe** [comment]
Right now I'm going to lunch. I will do in half an hour.

**Comment by steffanson** [comment]
> We test Vivaldi, Firefox, Chrome on Linux (partially on Win) as well as Safari on Mac during development, all in their most recent releases (regular releases, not development releases) and haven't seen such an issue.

I had the very same issue yesterday after updating from 3.1.4 to 3.2 with the FF 57.0.1 and Chromium 62.0.3202.94 on Linux. 

The error console did not reveal any problems to me; however what helped in my case was flushing the logs (on the terminal, as the button on the website is broken right now).  After doing that, it took about a minute until the statistics came back.

@JobbeDeluxe: Maby you can test it by moving your old logs to a backup folder?

**Comment by JobbeDeluxe** [comment]
I tested on FF, this will display the Dashboard, longterm data is not avaliable, but this can be a problem of the log flush maybe.
On Chrome is no error in the console
![auswahl_002](https://user-images.githubusercontent.com/20384365/33804736-d621889e-ddab-11e7-82ff-e94534715526.png)

**Comment by DL6ER** [comment]
What does http://pi.hole/admin/api.php tell you?

**Comment by JobbeDeluxe** [comment]
{"domains_being_blocked":106012,"dns_queries_today":4547,"ads_blocked_today":2420,"ads_percentage_today":53.221905,"unique_domains":302,"queries_forwarded":1174,"queries_cached":951,"clients_ever_seen":18,"unique_clients":18,"status":"enabled"}


Seems to be OK

**Comment by DL6ER** [comment]
> Seems to be OK

I agree. Could you try executing the following in your Console while you are on the dashboard?
```
updateSummaryData(true);
```

**Comment by JobbeDeluxe** [comment]
->updateSummaryData(true);
<-undefined

that what i get.
but the same i will get on FF

**Comment by Mcat12** [comment]
Did the summary stats update?

**Comment by JobbeDeluxe** [comment]
Nothing changed

**Comment by Mcat12** [comment]
What versions of browsers are you using?

**Comment by teapleasenosugar** [comment]
I had this problem. To fix, you have to clear the cache in safari. Remember that iCloud will sync this, I cleared the cache (for my piholes IP address) from Mac OS safari, reloaded the page and all is well.

**Comment by JobbeDeluxe** [comment]
I cleared my cache on mobile device and reloaded the page. This fixed the problem on my mobile, so I think it will do on laptop as well

**Comment by adi2k5** [comment]
I had the very same problem with data not showing, but it turned out that graphs were not displayed due to SELinux, after fixing this graphs are OK. But I can't get stats to work, clearing caches etc is not helping at all, all what I get is "---" for each field. When I'll go to api.php, it gives 
```
{"domains_being_blocked":2847358,"dns_queries_today":10331,"ads_blocked_today":4512,"ads_percentage_today":43.674377,"unique_domains":1024,"queries_forwarded":4606,"queries_cached":1172,"clients_ever_seen":1,"unique_clients":1,"status":0}
```
One thing that I noticed that might be causing the issue is this warning in php log: 

PHP Warning:  number_format() expects parameter 1 to be float, string given in /var/www/html/admin/api_FTL.php on line 39

**Comment by vivasmaq** [comment]
This morning i updated my version to the last one and i had this same dashboard blank(---) indicators on opera browser. I enter this issue and then reload the dashboard with ctrl+f5. Nothing changed. Luckly I have a firefox that i never use installed on a computer next to me. Work Great. I confirm, this is a client side browser cache related issue.

**Comment by Mcat12** [comment]
@adi2k5 Have you modified that file, or are you on the latest version? It does not appear to call `number_format`:
https://github.com/pi-hole/AdminLTE/blob/master/api_FTL.php#L39

**Comment by adi2k5** [comment]
@Mcat12 No, I haven't. To be sure I just wget copy from github and it's still the same. I've done some testing, and it seems that this warning comes from https://github.com/pi-hole/AdminLTE/blob/master/api_FTL.php#L55, as it stopped when I removed 'floatval()' from that line.

**Comment by Mcat12** [comment]
What is the output of this command?
```
echo ">stats" | nc localhost 4711
```

**Comment by adi2k5** [comment]
```
echo ">stats" | nc localhost 4711
domains_being_blocked 2852587
dns_queries_today 8627
ads_blocked_today 3677
ads_percentage_today 42.622002
unique_domains 983
queries_forwarded 4050
queries_cached 876
clients_ever_seen 1
unique_clients 1
status enabled
---EOM---
```

**Comment by DL6ER** [comment]
@adi2k5 that looks fine at first glance, can you quote the error you have seen in `lighttpd`'s log file?

**Comment by adi2k5** [comment]
That's the point, there's no errors in logs, just php warning mentioned above
```
[17-Dec-2017 11:36:16 Europe/Warsaw] PHP Warning:  number_format() expects parameter 1 to be float, string given in /var/www/html/admin/api_FTL.php on line 39
```
Except this warning I don't see other errors nor problems, access.log have only this
```
192.168.1.1 my.domain user [17/Dec/2017:11:58:50 +0100] "GET /admin/api.php?summary HTTP/1.1" 200 174 "https://my.domain/admin/index.php?login" "Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/62.0.3202.97 Safari/537.36 Vivaldi/1.94.1008.34"
192.168.1.1 my.domain user [17/Dec/2017:11:58:51 +0100] "GET /admin/api.php?summaryRaw&topItems HTTP/1.1" 200 325 "https://my.domain/admin/index.php?login" "Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/62.0.3202.97 Safari/537.36 Vivaldi/1.94.1008.34"
192.168.1.1 my.domain user [17/Dec/2017:11:58:51 +0100] "GET /admin/api.php?summaryRaw&getQuerySources HTTP/1.1" 200 197 "https://my.domain/admin/index.php?login" "Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/62.0.3202.97 Safari/537.36 Vivaldi/1.94.1008.34"
192.168.1.1 my.domain user [17/Dec/2017:11:58:51 +0100] "GET /admin/api.php?summary HTTP/1.1" 200 175 "https://my.domain/admin/index.php?login" "Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/62.0.3202.97 Safari/537.36 Vivaldi/1.94.1008.34"
```
Strange is that all is working as it supposed to work, charts, stats of top blocked domains etc. All except this 4 'small blocks'.

**Comment by DL6ER** [comment]
@adi2k5 Please open a separate issue on pi-hole/AdminLTE to keep things clear in here, I suggest to use the title `number_format() expects parameter 1 to be float, string given`

One question you could already answer therein is: How do you format numbers in Poland? E.g. we use `.` as decimal separator in English, but `,` in German. This might be a locale issue within `PHP` we haven't thought about yet.

**Comment by DL6ER** [comment]
@JobbeDeluxe @vivasmaq Is your issue resolved by now?

**Comment by JobbeDeluxe** [comment]
Clear the cache have solved the problem on all my devices

**Comment by adi2k5** [comment]
We use comma `,` as decimal separator, and thin space as thousands groups separator, this examples of fortmatting in some countries might be useful: https://docs.oracle.com/cd/E19455-01/806-0169/overview-9/index.html

**Comment by dschaper** [comment]
@adi2k5 Please open a new issue as DL6ER has requested.
